app/src/main/java/com/example/grassroots/network/CivicInfoRetrofit.java

```diff
@@ -0,0 +1,66 @@
+package com.example.grassroots.network;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.example.grassroots.MainActivity;
+import com.example.grassroots.R;
+import com.example.grassroots.model.CivicInfoModel;
+import com.example.grassroots.utils.Constant;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class CivicInfoRetrofit {
+
+    private static Retrofit retrofit;
+    private static CivicInfoRetrofit thisInstance;
+    private CivicInfoListener listener;
+
+    private CivicInfoRetrofit() {}
+
+    static CivicInfoRetrofit getInstance() {
+        if (thisInstance == null) {
+            thisInstance = new CivicInfoRetrofit();
+        }
+        return thisInstance;
+    }
+
+    private CivicInfoService onConnect() {
+        if (retrofit == null) {
+            retrofit = new Retrofit.Builder()
+                    .baseUrl(Constant.KEYS.CIVIC_INFO_BASE_URL)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build();
+        }
+        return retrofit.create(CivicInfoService.class);
+    }
+
+    void onSuccess(String inputAddress, Context context) {
```

Comment:
For proper isolation you should not pass in Context to a method in your retrofit or repository

Reply:
You should also pass in a listener per network request not set it on the class instance